packages/belos/tpetra/test/MINRES/test_pminres_hb.cpp

```diff
@@ -0,0 +1,252 @@
+    cmdp.setOption("filename",&filename,"Filename for Harwell-Boeing test matrix.");
+    cmdp.setOption("tol",&tol,"Relative residual tolerance used by Minres solver.");
+    cmdp.setOption("num-rhs",&numrhs,"Number of right-hand sides to be solved for.");
+    cmdp.setOption("max-iters",&maxiters,"Maximum number of iterations per linear system (-1 := adapted to problem/block size).");
+
+    if (cmdp.parse(argc,argv) != Teuchos::CommandLineProcessor::PARSE_SUCCESSFUL) {
+      return -1;
+    }
+    if (!verbose)
+      frequency = -1;  // Reset frequency if verbosity is off
+
+    proc_verbose = ( verbose && (MyPID==0) );
+    if (proc_verbose) {
+      std::cout << Belos::Belos_Version() << std::endl << std::endl;
+    }
+
+    //
+    // Get the problem
+    //
+    RCP<tcrsmatrix_t> A;
+    Tpetra::Utils::readHBMatrix(filename,comm,A);
+    RCP<const Tpetra::Map<> > map = A->getDomainMap();
+
+    // Create initial vectors
+    RCP<tmultivector_t> X, B;
+    X = rcp(new MV(map, numrhs));
+    MVT::MvRandom( *X );
+    B = rcp(new MV(map, numrhs));
+    OPT::Apply(*A, *X, *B);
+    MVT::MvInit(*X, 0.0);
+    
+    //
+    // *****Create parameter list for the Minres solver manager*****
+    //
+    const int NumGlobalElements = B->getGlobalLength();
+    if (maxiters == -1)
+      maxiters = NumGlobalElements - 1; // maximum number of iterations to run
+    //
+    ParameterList belosList;
+    belosList.set( "Maximum Iterations", maxiters );       // Maximum number of iterations allowed
+    belosList.set( "Convergence Tolerance", tol );         // Relative convergence tolerance requested
+    belosList.set( "Flexible Gmres", true );               // use FGMRES
+                                                           
+    int verbLevel = Belos::Errors + Belos::Warnings;
+    if (verbose) {
+      verbLevel += Belos::TimingDetails + Belos::FinalSummary + Belos::StatusTestDetails;
+    }
+    belosList.set( "Verbosity", verbLevel );
+    if (verbose) {
+      if (frequency > 0) {
+        belosList.set( "Output Frequency", frequency );
+      }
+    }
+
+    // Set parameters for the inner GMRES (preconditioning) iteration.
+    ParameterList innerBelosList;
+    innerBelosList.set( "Maximum Iterations", maxiters );       // Maximum number of iterations allowed
+    innerBelosList.set( "Convergence Tolerance", tol );         // Relative convergence tolerance requested
+    innerBelosList.set( "Verbosity", Belos::Errors + Belos::Warnings );
+
+    // *****Construct linear problem for the inner iteration using A *****
+    Belos::LinearProblem<ST,MV,OP> innerProblem;
+    innerProblem.setOperator( A );
+    innerProblem.setLabel( "Belos Preconditioner Solve" );
+
+    //  
+    // *****Create the inner block Gmres iteration********
+    //  
+    RCP<Belos::TpetraOperator<ST>> innerSolver;
+    innerSolver = rcp( new Belos::TpetraOperator<ST>( rcpFromRef(innerProblem), rcpFromRef(innerBelosList), "Block Gmres", true ) );
```

Comment:
Probably best, for mathematical reasons, to make this "Minres" again instead of "Block Gmres"